### PR TITLE
Fixes #20415: The technique editor allows saving methods with missing parameters

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/20415
I add a little feedback to see wich method's parameters are mandatory
image

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/ViewTechnique.elm [hunk 1]
```diff
@@ -60,6 +61,48 @@ showTechnique model technique origin ui =
...
+                let
+                  paramConstraints = case (List.head (List.filter (\p -> param.id.value == p.name.value) method.parameters)) of
+                    Just paramWithConstraints -> paramWithConstraints.constraints
+                    _                         -> []
+                  state = accumulateErrorConstraint param paramConstraints ValidState
+
+                in
+                (mCall.id.value, state)
+              )  mCall.parameters
+            _      -> [("hehehe",ValidState)]
```
Comment (ElaadF) [inline webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/ViewTechnique.elm:101]: change `hehehe` by `mCall.id.value`

File: webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/ViewTechnique.elm [hunk 2]
```diff
@@ -60,6 +61,48 @@ showTechnique model technique origin ui =
...
+             - [Optional] with parameter  "Version"      -> "latest"
+             - [Optional] with parameter  "Architecture" -> empty
+             - [Optional] with parameter  "Provider"     -> "apt"
+
+       the result will be :
+       [
+         ( "e8a8d662-9d2d-44b4-87f8-591fb6f4db1b"
+         , InvalidState [ConstraintError { id = { value = "command" }, message = "Parameter 'command' is empty" }])
+         ) ,
+         ( "e8a8d662-9d2d-44b4-87f8-591fb6f4db1b"
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/ViewTechnique.elm:82]: shouldn' it be 6394770e-a9c2-4e67-8569-d21560a07dc2 and parameter Name is empty ?

File: webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/ViewTechnique.elm (hunk 1 above)
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/ViewTechnique.elm:101]: i'm unsure about that ?

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Comment by ElaadF** [comment]
OK, squash merging this PR

**Pull request merged**